migrations/0002_upgrade_materialize.py

```diff
@@ -0,0 +1,148 @@
+# Generated by Django 3.2.19 on 2023-06-05 15:33
+
+import os
+import sys
+import difflib
+
+from django.db import migrations
+from django.conf import settings
+from plugins.customstyling import plugin_settings
+
+from utils.logger import get_logger
+logger = get_logger(__name__)
+
+
+def change_old_to_new_css_selectors(file_paths, reverse=False):
+    """
+    Modifies custom stylesheets so that they
+    use the new selectors in v1 and later of Materialize CSS.
+    See https://github.com/Dogfalo/materialize/blob/v1-dev/v1-upgrade-guide.md
+    """
+    changes_to_make = [
+        ('dropdown-button', 'dropdown-trigger'),
+        ('button-collapse', 'sidenav-trigger'),
+        ('side-nav', 'sidenav'),
+        ('data-activates', 'data-target'),
+    ]
+
+    for file_path in file_paths:
+        try:
+            css = []
+            new_css = []
+            with open(file_path, 'r', encoding="utf-8") as file_ref:
+                css = file_ref.readlines()
+                for line in css:
+                    for change in changes_to_make:
+                        if reverse:
+                            new, old = change
+                        else:
+                            old, new = change
+                        line = line.replace(old, new)
+                    new_css.append(line)
+            if css != new_css:
+                with open(file_path, 'w', encoding="utf-8") as file_ref:
```

Comment:
Can we backup the file before we do this? It'd make it a bit simpler to revert if anything goes amiss beyond the logic on the migration.

Reply:
Sure. I have just added a backup.